test/controllers/works_controller_test.rb

```diff
@@ -2,115 +2,280 @@
+    end
 
+    it "A logged in user can...renders 400 bad_request for bogus categories" do
+
+      perform_login(person)
+      work = {
+          another_movie: {
+            title: "Sound of Music",
+            category: "popcorn_head"
+          }
+        }
+      post works_path, params: work
+      must_respond_with :bad_request
     end
 
   end
 
   describe "show" do
-    it "succeeds for an extant work ID" do
-
-    end
 
-    it "renders 404 not_found for a bogus work ID" do
+    let(:person){users(:june)}
 
+    it "A non-logged in user cannot access view templates associated with the show action" do
+      get works_path
+      must_respond_with :redirect
     end
-  end
-
-  describe "edit" do
-    it "succeeds for an extant work ID" do
 
+    it "A logged in user can...succeeds for an extant work ID" do
+      perform_login(person)
+      work = works(:album)
+      get work_path(work.id)
+      must_respond_with :success
     end
 
-    it "renders 404 not_found for a bogus work ID" do
+    it "A logged in user can...renders 404 not_found for a bogus work ID" do
 
+      perform_login(person)
+      work = works(:album)
+      work.id = 0
+      get work_path(work.id)
+      must_respond_with :not_found
     end
   end
 
-  describe "update" do
-    it "succeeds for valid data and an extant work ID" do
+  describe "edit" do
 
-    end
+    let(:person){users(:june)}
 
-    it "renders bad_request for bogus data" do
+    it "A non-logged in user cannot access view templates associated with the edit action" do
+      get works_path
+      must_respond_with :redirect
+    end
 
+    it "A logged in user can...succeeds for an extant work ID" do
+      perform_login(person)
+      id = works(:album)
+      get edit_work_path(id)
+      must_respond_with :success
     end
 
-    it "renders 404 not_found for a bogus work ID" do
+    it "A logged in user can...renders 404 not_found for a bogus work ID" do
 
+      perform_login(person)
+      work = works(:album)
+      work.id = 0
+      get edit_work_path(work.id)
+      must_respond_with :not_found
     end
   end
 
-  describe "destroy" do
-    it "succeeds for an extant work ID" do
+  describe "update" do
+    let(:person){users(:june)}
 
+    it "A non-logged in user cannot access view templates associated with the update action" do
+      get works_path
+      must_respond_with :redirect
     end
 
-    it "renders 404 not_found and does not update the DB for a bogus work ID" do
-
-    end
+#     it "a logged in user can...succeeds for valid data and an extant work ID" do
+#
+#       perform_login(person)
+#
+#       work_to_change = works(:poodr)
+# # binding.pry
+#       new_work_title = "99 Bottles"
+# # binding.pry
+#       work = {
+#         work: {
+#           title: new_work_title
+#         }
+#       }
+# # binding.pry
+#
+#       patch work_path(work_to_change.id), params[:work]
+# # binding.pry
+#       expect(work_to_change.title).must_equal new_work_title
```

Comment:
This is almost there. There are two things that are missing:

- When you add the `params` parameter to the `patch` method, you should do it as a keyword argument:
    ```ruby
    patch work_path(work_to_change.id), params: work
    ```
- You need to reload `work_to_change` from the database before the expectation on line 217: `work_to_change.reload`. `work_to_change` is a local variable that was loaded _before_ you made the change, which means if the DB is updated it won't reflect the change.